cs/Markdown/Markdown/TagItalic.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Collections.Generic;
+
+namespace Markdown
+{
+    public class TagItalic : ITag
+    {
+        public bool IsAtTheBeginning { get; set; }
+        public bool IsClosed { get; set; }
+
+        private string HtmlTagAnalog
+        {
+            get
+            {
+                if (IsClosed)
+                    return (IsStartTag) ? "<em>" : "</em>";
+                return "_";
+            }
+        }
+
+        public bool IsStartTag { get; set; }
+
+        public string Content => HtmlTagAnalog;
+
+        public bool IsNotToPairToken { get; set; }
+
+        public void GenerateProperties(LinkedListNode<IToken> currentToken)
+        {
+            var spacesCnt = 0;
+            var boldCnt = 0;
+            var onlyEmptyStrings = true;
+            while (currentToken != null)
+            {
+                if (currentToken.Value is TagSpace)
+                    spacesCnt++;
+                if (currentToken.Value is TagBold)
+                    boldCnt++;
+                if (currentToken.Value is TagBold && currentToken.Value.IsNotToPairToken && boldCnt % 2 != 0)
+                    break;
+                if (currentToken.Value is TagItalic && !currentToken.Value.IsNotToPairToken)
+                {
+                    if (currentToken.Value is ITag starter)
```

Comment:
Кажется, этот `if` всегда `true`?

Reply:
Да, он должен быть верным, но это нужно, чтобы сама среда разработки вместе с решарпером не ругалась на возможный nullReferenceException.
 Стоит ли убрать?

Reply:
Есть несколько способов это обработать, и тот что ты выбрал - наименее понятный.

Давай отсюда уберём starter, и добавим на две строки выше
```suggestion
                    if (currentToken.Value is ITag)
```